projects/storefrontlib/cms-components/cart/validation/cart-validation-state.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { ReplaySubject, Subscription } from 'rxjs';
+import { take, tap, withLatestFrom } from 'rxjs/operators';
+import { RoutingService, CartModification } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartValidationStateService implements OnDestroy {
+  NAVIGATION_SKIPS = 2;
+  navigationIdCount = 0;
+
+  private subscription = new Subscription();
+  cartValidationResult$ = new ReplaySubject<CartModification[]>(1);
```

Comment:
It should be visible to the outside as `Observable`, to avoid calls of `next` from outside.
```suggestion
  cartValidationResult$: Observable<CartModification[]> = new ReplaySubject<CartModification[]>(1) as Observable<CartModification[]>;
```